base/shell/explorer/trayclock.cpp

```diff
@@ -529,19 +533,19 @@ LRESULT CTrayClockWnd::OnPaint(UINT uMsg, WPARAM wParam, LPARAM lParam, BOOL& bH
         rcClient.top = (rcClient.bottom - CurrentSize.cy) / 2;
         rcClient.bottom = rcClient.top + CurrentSize.cy;
 
-        for (i = 0, line = 0;
-                i < CLOCKWND_FORMAT_COUNT && line < VisibleLines;
-                i++)
+        if (VisibleLines == 2 && !g_TaskbarSettings.bShowWeekday)
         {
-            if (LineSizes[i].cx != 0)
-            {
-                TextOut(hDC,
-                    (rcClient.right - LineSizes[i].cx) / 2,
-                    rcClient.top + TRAY_CLOCK_WND_SPACING_Y,
-                    szLines[i],
-                    wcslen(szLines[i]));
+            PaintLine(hDC, &rcClient, 0, CLOCKWND_FORMAT_TIME);
+            PaintLine(hDC, &rcClient, 1, CLOCKWND_FORMAT_DATE);
```

Comment:
Why would he use that, unless that specific case gets merged with the for-loop from below?
Because at it can be _**explicitly**_ seen in the if-condition above, that if-block is taken in particular when `bShowWeekday` is FALSE. Thus, your code you are proposing would always take the "FALSE" branch, namely use the `CLOCKWND_FORMAT_DATE` in all cases when that if-branch is taken.

Reply:
> unless that specific case gets merged with the for-loop from below?

That additional behavior change (which I am not supporting) was not requested, but it could be done if people do want it, since this PR is ReactOS specific.

>  the if-condition above

Indeed, my suggestion implies (re)moving that new condition and uncommenting out the other define...

Reply:
ok then I think you should give the whole alternative code you are thinking about, because it's not fully yet clear (at least to me) what you have in mind.

Reply:
I tried to make the fewest changes to the code as possible, hence the design here. It is my opinion that this would be less readable than what I already wrote. I think the only way this makes sense is if I were to replace the for loop with if statements for one, two, and three lines; which would be more code to review.

Reply:
> whole alternative code

Suggested.

Note:
my comments about reversed default and lack of documentation stands.

Reply:
Ok I think Serge's code variant here is simpler to understand, I will apply it.